### DISCUSSION TITLE
Linux packaging using NixOS

### DISCUSSION THREAD

**Discussion by lobre** [athensresearch/athens#856] [Feature Requests]
Hi,

I don't know how many people are in the same boat as me, but I am running Linux and more precisely NixOS.
NixOS is really solid for having a declarative version of all your configurations and dependencies.

As of today, I am not sure there has been a lot of thoughts regarding packaging in Linux apart from the currently provided AppImage format. I have searched through Github issues, discussions, and the Discord history.

But I guess many Linux users prefer installing software using their preferred package manager instead of installing each application on its own. It allows precisely to upgrade the system as a whole.

However, Linux package managers usually don't really play well with "self-updaters" (as implemented for instance in Athens). I understand why it exists, especially in non-Linux environments, but I would personally prefer to have a possibility to disable it and handle the Athens upgrade lifecycle using my package manager of choice.

So this discussion would be about two points:
- Add a configuration in Athens to allow to disable self-update.
- Think about packaging Athens using famous Linux package managers.

For this second point, I would personally be interested in the Nix package manager. Nix is performing better when the application is built from sources, but it also has a way to built taking an AppImage as an entry (which could be a nice first-step).

And now that Athens is distributed using Github releases, it simplifies also the packaging (as many package managers can take input from Github releases).

I'll be interested in knowing the opinion of other Linux users.

And I might give it a try to package in Nix starting from the AppImage. Self-update would then probably fail because the application would be installed in a read-only filesystem. But if it fails silently, that could do at the moment.

Thanks,
Loric

**Comment by lobre** [comment]
This could help to resolve this issue also: https://github.com/athensresearch/athens/issues/706

**Reply by westofer** [reply]
btw when running athens from AUR (athens-git), I get this error/warning

```
20:21:37.431 › Error: Error: ENOENT, dev-app-update.yml not found in /usr/lib/athens.asar

```

so I assume it won't auto update

full error log: https://pastebin.com/SVtw51Us

**Comment by jefftangx** [comment]
Really wanna try NixOS. Very likely could make update more configurable, but slightly harder because that's an electron setting and not front-end.

**Reply by lobre** [reply]
Could it be for instance a CLI parameter such as `--no-auto-update` or something like this? But yes, effectively, I understand this setting has nothing to do with the frontend application per se.
